File: backend/src/main/java/ty/tran/demo/Entity/Call.java
package ty.tran.demo.Entity;

import java.time.Instant;
import java.util.UUID;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.*;

@Entity
@Table(name = "calls")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Call {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "conversation_id")
    private Conversation conversation;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "initiator_id")
    private User initiator;

    @Enumerated(EnumType.STRING)
    private CallType type;

    @Enumerated(EnumType.STRING)
    private CallStatus status = CallStatus.RINGING;

    private Instant startedAt;
    private Instant endedAt;
    private Instant createdAt;

    public enum CallType {
        AUDIO, VIDEO
    }

    public enum CallStatus {
        RINGING, ONGOING, ENDED, MISSED, CANCELED, FAILED
    }
}

File: frontend/src/types/call.ts
export interface CallDTO {
  id: string;
  conversationId: string;
  initiatorId: string;
  initiatorName: string;
  initiatorEmail: string;
  type: 'audio' | 'video';
  status: 'ringing' | 'ongoing' | 'ended' | 'missed' | 'canceled' | 'failed';
  startedAt?: string;
  endedAt?: string;
  createdAt: string;
  duration?: number;
  conversationTitle: string;
}

export interface CreateCallRequest {
  conversationId: string;
  type: 'audio' | 'video';
}

export interface UpdateCallStatusRequest {
  status: 'ringing' | 'ongoing' | 'ended' | 'missed' | 'canceled' | 'failed';
}

